shared/middleware/net_sim_api.rb

```diff
@@ -237,85 +237,82 @@ def owns_shard?(shard_id)
     result.to_json
   end
 
+  def validate_all(shard_id, table_name, values)
+    values.map { |value| validate_one(shard_id, table_name, value) }
+  end
+
   # @param [String] shard_id - The shard we're checking validation on.
   # @param [String] table_name - The table we're validating for
   # @param [Hash] value - The value we're validating
-  # @return [Boolean] True if the new value appears to be well-formed.
-  def value_valid?(shard_id, table_name, value)
+  # @return [String] a validation error, or nil if no problems were found
+  def validate_one(shard_id, table_name, value)
+    return INSERT_ERRORS[:malformed] unless value.is_a? Hash
```

Comment:
Yeah, I guess I should do this now since we'll eventually want errors for the other operations.